dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -145,11 +150,43 @@ public Entry addAliases(@ApiParam(hidden = true) @Parameter(hidden = true, name
     public List<VersionVerifiedPlatform> getVerifiedPlatforms(@Parameter(hidden = true, name = "user")@Auth Optional<User> user,
             @Parameter(name = "entryId", description = "id of the entry", required = true, in = ParameterIn.PATH) @PathParam("entryId") Long entryId) {
         Entry<? extends Entry, ? extends Version> entry = toolDAO.getGenericEntryById(entryId);
-        checkOptionalAuthRead(user, entry);
+        checkEntry(entry);
+
+        if (!entry.getIsPublished()) {
+            checkUser(user.get(), entry);
+        }
+
         List<VersionVerifiedPlatform> verifiedVersions = versionDAO.findEntryVersionsWithVerifiedPlatforms(entryId);
         return verifiedVersions;
     }
 
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/{entryId}/versions/{versionId}/fileTypes")
+    @ApiOperation(value = "Retrieve the file types of a version's sourcefiles",  hidden = true)
+    @Operation(operationId = "getVersionsFileTypes", description = "Retrieve the file types of a version's sourcefile", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public SortedSet<DescriptorLanguage.FileType> getVersionsFileTypes(@Parameter(hidden = true, name = "user")@Auth Optional<User> user,
+            @Parameter(name = "entryId", description = "Entry to retrieve the version from", required = true, in = ParameterIn.PATH) @PathParam("entryId") Long entryId,
+            @Parameter(name = "versionId", description = "Version to retrieve the sourcefile types from", required = true, in = ParameterIn.PATH) @PathParam("versionId") Long versionId) {
+        Entry<? extends Entry, ? extends Version> entry = toolDAO.getGenericEntryById(entryId);
+        checkEntry(entry);
+
+        if (!entry.getIsPublished()) {
+            checkUser(user.get(), entry);
+        }
+
+        Version version = versionDAO.findVersionInEntry(entryId, versionId);
+        if (version == null) {
+            throw new CustomWebApplicationException("Version " + versionId + " does not exist for this entry", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        SortedSet<DescriptorLanguage.FileType> fileTypes = new TreeSet<>();
```

Comment:
There's no reason to create a TreeSet here, because you overwrite it on line 186. I would just get rid of this line and define it on line 186. 

Or, more subjectively, not not declare it at all and just return line 186 directly.